File: wrappers/fedimint-ts/types/modules/ln.ts
interface LnInvoiceRequest {
    amountMsat: number;
    description: string;
    expiryTime?: number;
}

interface LnInvoiceResponse {
    operationId: string;
    invoice: string;
}

interface AwaitInvoiceRequest {
    operationId: string;
}

interface LnPayRequest {
    paymentInfo: string;
    amountMsat?: number;
    finishInBackground: boolean;
    lnurlComment?: string;
}

interface LnPayResponse {
    operationId: string;
    paymentType: string;
    contractId: string;
    fee: number;
}

interface AwaitLnPayRequest {
    operationId: string;
}

interface Gateway {
    nodePubKey: string;
    active: boolean;
}

interface ListGatewaysResponse {
    [federationId: string]: Gateway[];
}

interface SwitchGatewayRequest {
    gatewayId: string;
}

export type {
    LnInvoiceRequest,
    LnInvoiceResponse,
    AwaitInvoiceRequest,
    LnPayRequest,
    LnPayResponse,
    AwaitLnPayRequest,
    Gateway,
    ListGatewaysResponse,
    SwitchGatewayRequest,
}

File: wrappers/fedimint-py/models/ln.py
from pydantic import BaseModel
from typing import Optional


class LnInvoiceRequest(BaseModel):
    amount_msat: int
    description: str
    expiry_time: Optional[int]


class LnInvoiceResponse(BaseModel):
    operation_id: str
    invoice: str


class AwaitInvoiceRequest(BaseModel):
    operation_id: str


class LnPayRequest(BaseModel):
    payment_info: str
    amount_msat: Optional[int]
    finish_in_background: bool
    lnurl_comment: Optional[str]


class LnPayResponse(BaseModel):
    operation_id: str
    payment_type: str
    contract_id: str
    fee: int


class AwaitLnPayRequest(BaseModel):
    operation_id: str


class Gateway(BaseModel):
    node_pub_key: str
    active: bool


class SwitchGatewayRequest(BaseModel):
    gateway_id: str
